### PR TITLE
 xsns_33_qmc5883l.ino check for overflow  and scale reading

### PR DESCRIPTION
**Pull request by fb-pilot**
## Description:
xsns_33_qmc5883l.ino Add cofigure some chip-parameters, check for overflow  and scale the reading to tesla
**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by fb-pilot** [comment]
now I checked all Tasmota cores :-))

File: tasmota/my_user_config.h [hunk 1]
```diff
@@ -470,7 +470,8 @@
   #define USE_ENHANCED_GUI_WIFI_SCAN             // Enable Wi-Fi scan output with BSSID (+0k5 code)
 //  #define USE_WEBSEND_RESPONSE                   // Enable command WebSend response message (+1k code)
 //  #define USE_WEBGETCONFIG                       // Enable restoring config from external webserver (+0k6)
-//  #define USE_GPIO_VIEWER                        // Enable GPIO Viewer to see realtime GPIO states (+6k code)
+//  #define USE_GPIO_VIEWER                        // Enable GPIO Viewer to see realtime GPIO states (+4k code)
+//    #define GV_BASE_URL           "https://thelastoutpostworkshop.github.io/microcontroller_devkit/gpio_viewer/assets/"
```
Comment (arendst) [inline tasmota/my_user_config.h:474]: Pls rebase your PR as this line has been removed

Comment (fb-pilot) [reply tasmota/my_user_config.h:475]: done - I hope thats what you need  :-))

Comment (fb-pilot) [reply tasmota/my_user_config.h:475]: I am confused ...
The "my_user_config.h" in my fork is only extended with some #define USE_QMC58...
I see the identical #define USE_GPIO_VIEWER  in both "my_user_config.h" (in my fork and in the original development in two lines :
474 ... //  #define USE_GPIO_VIEWER                        // Enable GPIO Viewer to see realtime GPIO states (+4k code)
and
1105 ... #define USE_GPIO_VIEWER                          // Enable GPIO Viewer to see realtime GPIO states (+5k6 code)
line 475 is ...
475 ... //    #define GV_SAMPLING_INTERVAL  100            // [GvSampling] milliseconds - Use Tasmota Scheduler (100) or Ticker (20..99,101..1000)
??? what can I do ???

Comment (fb-pilot) [reply tasmota/my_user_config.h:475]: I  rebased my PR on dev[13.4.0.1] now 
I hope it works now

Comment (arendst) [reply tasmota/my_user_config.h:475]: No it doesn't. You should be able to easily see that it fails on line 475. That line is not present in dev or released branch.

So keep trying until you see for yourself that line 475 is gone.

Comment (arendst) [reply tasmota/my_user_config.h:475]: I'll fix it.

**Comment by arendst** [comment]
Thx. It's merged now.

**Comment by fb-pilot** [comment]
Thx 👍  ... I still dont check what the problem was !? :-((

**Pull request closed without merging**